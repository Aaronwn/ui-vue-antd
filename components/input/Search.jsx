
import classNames from 'classnames'
import Input from './Input'
import Icon from '../icon'
import inputProps from './inputProps'
import Button from '../button'
import { cloneElement } from '../_util/vnode'
import { getOptionProps, getComponentFromProp } from '../_util/props-util'
import PropTypes from '../_util/vue-types'

export default {
  name: 'AInputSearch',
  props: {
    ...inputProps,
    prefixCls: {
      default: 'ant-input-search',
      type: String,
    },
    inputPrefixCls: {
      default: 'ant-input',
      type: String,
    },
    enterButton: PropTypes.oneOfType([PropTypes.bool, PropTypes.string, PropTypes.object]),
  },
  model: {
    prop: 'value',
    event: 'change.value',
  },
  methods: {
    onSearch (e) {
      this.$emit('search', this.$refs.input.stateValue)
      this.$refs.input.focus()
    },
    focus () {
      this.$refs.input.focus()
    },

    blur () {
      this.$refs.input.blur()
    },
    getButtonOrIcon () {
      const { prefixCls, size } = this
      const enterButton = getComponentFromProp(this, 'enterButton')
      if (!enterButton) {
        return <Icon class={`${prefixCls}-icon`} type='search' key='searchIcon' />
      }
      const enterButtonAsElement = Array.isArray(enterButton) ? enterButton[0] : enterButton
      if (enterButtonAsElement.componentOptions && enterButtonAsElement.componentOptions.Ctor.extendOptions.__ANT_BUTTON) {
        return cloneElement(enterButtonAsElement, {
          class: `${prefixCls}-button`,
          props: { size },
          on: {
            click: this.onSearch,
          },
        })
      } else if (enterButtonAsElement.tag === 'button') {
        return cloneElement(enterButtonAsElement, {
          on: {
            click: this.onSearch,
          },
        })
      }
      return (
        <Button
          class={`${prefixCls}-button`}
          type='primary'
          size={size}
          onClick={this.onSearch}
          key='enterButton'
        >
          {enterButton === true ? <Icon type='search' /> : enterButton}
        </Button>
      )
    },
  },
  render () {
    const { prefixCls, inputPrefixCls, size,
      suffix, ...others
    } = getOptionProps(this)
    const enterButton = getComponentFromProp(this, 'enterButton')
    const buttonOrIcon = this.getButtonOrIcon()
    const searchSuffix = suffix ? [suffix, buttonOrIcon] : buttonOrIcon
    const inputClassName = classNames(prefixCls, {
      [`${prefixCls}-enter-button`]: !!enterButton,
      [`${prefixCls}-${size}`]: !!size,
    })
    const inputProps = {
      props: {
        ...others,
        prefixCls: inputPrefixCls,
        size,
        suffix: searchSuffix,
      },
      attrs: this.$attrs,
      on: {
        ...this.$listeners,
        pressEnter: this.onSearch,
      },
    }
    return (
      <Input
        {...inputProps}
        class={inputClassName}
        ref='input'
      />
    )
  },
}

