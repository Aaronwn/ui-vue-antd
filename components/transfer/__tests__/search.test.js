import { mount } from '@vue/test-utils'
import Search from '../search'

describe('Search', () => {
  it('should show cross icon when input value exists', () => {
    const props = {
      propsData: {
        value: '',
      },
    }
    const wrapper = mount(Search, props)

    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({ value: 'a' })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
