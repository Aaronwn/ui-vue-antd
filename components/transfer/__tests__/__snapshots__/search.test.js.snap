// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should show cross icon when input value exists 1`] = `
<div>
  <input value="" placeholder="" type="text" class="ant-input"><span class="undefined-action"><i class="anticon anticon-search"></i></span></div>
`;

exports[`Search should show cross icon when input value exists 2`] = `
<div>
  <input value="a" placeholder="" type="text" class="ant-input">
  <a href="#" class="undefined-action">
    <i class="anticon anticon-cross-circle"></i>
  </a>
</div>
`;
